agent/agentbootstrap/bootstrap.go

```diff
@@ -256,7 +256,7 @@ func (b *AgentBootstrap) Initialize(ctx stdcontext.Context) (_ *state.Controller
 		addAdminUser,
 		cloudbootstrap.InsertCloud(user.NameFromTag(b.adminUser), stateParams.ControllerCloud),
 		credbootstrap.InsertCredential(credential.KeyFromTag(cloudCredTag), cloudCred),
-		cloudbootstrap.SetCloudDefaults(stateParams.ControllerCloud.Name, stateParams.ControllerInheritedConfig),
+		modeldefaultsbootstrap.SetCloudDefaults(stateParams.ControllerCloud.Name, stateParams.ControllerCloud.Config),
```

Comment:
I'm not sure this is correct.
This I think will ignore any `--config` parameters pass by the user at bootstrap, which is what ControllerInheritedConfig is set to reflect. The ControllerCloud value would have any defaults set in the user's local cloud yaml I think (from memory), but not any values specified as args to bootstrap. 
ControllerInteritedConfig is used elsewhere as well, so just changing it here seems insufficient regardless. eg
```
	controllerModelDefaults := modeldefaultsbootstrap.ModelDefaultsProvider(
		stateParams.ControllerInheritedConfig,
		stateParams.RegionInheritedConfig[stateParams.ControllerCloudRegion],
		stateParams.ControllerCloud.Type,
	)
```

Reply:
> All fixed up. This is still broken in my opinion but an argument for another day.

The fact you disagree with the signed off design from 8 years ago doesn't mean it's broken :-)
It means you disagree with the product feature. Another day's discussion for sure.

Reply:
It isn't the design it is the implementation. The 3.6 code uses ControllerInheritedConfig to set a cloud model defaults and then to set the region defaults we use the cloud object in init params.

The diverge for no reasons and the defaults map we have setup on the cloud object also has space for cloud defaults that isn't being used. We aren't consistent in implementation or naming and so the contracts of what Juju is doing is both hard to grok and impossible to maintain.

It is broken due to know consistency in the model we have.